starter_code/src/movies.js

```diff
@@ -1,15 +1,115 @@
 /* eslint no-restricted-globals: 'off' */
 
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+function orderByYear(movies) {
+
+  // Shallow copy
+  let moviesYears = movies.map( (movie, i) => { return movie });
+
+  // Sort movies by year
+  moviesYears.sort( (a,b) => {
+      return a.year - b.year;
+  });
+
+  // Sort movies of the same year by title
+  moviesYears.sort( (a,b) => {
+    if (a.year === b.year) {
+      return a.title === b.title ? 0 : a.title > b.title ? 1 : -1;
+    }
+  });
+
+  return moviesYears;
+}
+
```

Comment:
Here you could chain everything at once, and especially, you can add more logic to your sort function.

moviesYears.sort((a,b) => {
if(a.year === b.year) return a..title.localeCompare(b.title)
return a.year - b.year;
}
Information on localeCompare: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

Reply:
Agreed, this one was very verbose for no reason !